dm/syncer/data_validator.go

```diff
@@ -954,7 +958,12 @@ func (v *DataValidator) processRowsEvent(header *replication.EventHeader, ev *re
 
 func (v *DataValidator) checkAndPersistCheckpointAndData(loc binlog.Location) error {
 	metaFlushInterval := v.cfg.ValidatorCfg.MetaFlushInterval.Duration
-	if time.Since(v.lastFlushTime) > metaFlushInterval {
+	cutOverLocation := v.cutOverLocation.Load()
+	needCutOver := cutOverLocation != nil && binlog.CompareLocation(*cutOverLocation, loc, v.cfg.EnableGTID) <= 0
```

Comment:
validator never validate before syncer's flushedPos, see waitSyncerSynced, maybe we should trigger syncer flush too in cutover

Reply:
Get your point. What about in next PR? It's hard to catch this now.

Reply:
lgtm